SmartDeviceLink/SDLSystemCapabilityManager.m

```diff
@@ -352,94 +232,140 @@ For backward compatibility (AppLink 2.0) static image type is always presented
-    return @[SDLSystemCapabilityTypeAppServices, SDLSystemCapabilityTypeNavigation, SDLSystemCapabilityTypePhoneCall, SDLSystemCapabilityTypeVideoStreaming, SDLSystemCapabilityTypeRemoteControl, SDLSystemCapabilityTypeDisplays, SDLSystemCapabilityTypeSeatLocation];
-}
-
 # pragma mark Subscribing
 
-/**
- * Sends a subscribe request for all possible system capabilites. If connecting to Core versions 4.5+, the requested capability will be returned in the response. If connecting to Core versions 5.1+, the manager will received `OnSystemCapabilityUpdated` notifications when the capability updates if the subscription was successful.
- */
-- (void)sdl_subscribeToSystemCapabilityUpdates {
-    for (SDLSystemCapabilityType type in [self.class sdl_systemCapabilityTypes]) {
-        SDLGetSystemCapability *getSystemCapability = [[SDLGetSystemCapability alloc] initWithType:type];
-        if (self.supportsSubscriptions) {
-            getSystemCapability.subscribe = @YES;
-        }
-
-        [self sdl_sendGetSystemCapability:getSystemCapability completionHandler:nil];
-    }
-}
+/// Sends a GetSystemCapability and sends back the response
+/// @param type The type to get
+/// @param subscribe Whether to change the subscription status. YES to subscribe, NO to unsubscribe, nil to keep whatever the current state is
+/// @param handler The handler to be returned
+- (void)sdl_sendGetSystemCapabilityWithType:(SDLSystemCapabilityType)type subscribe:(nullable NSNumber<SDLBool> *)subscribe completionHandler:(nullable SDLCapabilityUpdateWithErrorHandler)handler {
+    SDLLogV(@"Sending GetSystemCapability with type: %@, subscribe: %@", type, subscribe);
+    SDLGetSystemCapability *getSystemCapability = [[SDLGetSystemCapability alloc] initWithType:type];
+    getSystemCapability.subscribe = subscribe;
 
-/**
- *  Sends a `GetSystemCapability` to Core and handles the response by saving the returned data and notifying the subscriber.
- *
- *  @param getSystemCapability The `GetSystemCapability` request to send
- */
-- (void)sdl_sendGetSystemCapability:(SDLGetSystemCapability *)getSystemCapability completionHandler:(nullable SDLUpdateCapabilityHandler)handler {
-    __weak typeof(self) weakSelf = self;
+    __weak typeof(self) weakself = self;
     [self.connectionManager sendConnectionRequest:getSystemCapability withResponseHandler:^(__kindof SDLRPCRequest * _Nullable request, __kindof SDLRPCResponse * _Nullable response, NSError * _Nullable error) {
+        if (![response isKindOfClass:[SDLGetSystemCapabilityResponse class]]) {
+            SDLLogE(@"GetSystemCapability failed, type: %@, did not return a GetSystemCapability response", type);
+            if (handler == nil) { return; }
+            handler(nil, NO, [NSError sdl_systemCapabilityManager_moduleDoesNotSupportSystemCapabilities]);
+            return;
+        }
+
         if (error != nil) {
+            SDLLogE(@"GetSystemCapability failed, type: %@, error: %@", type, error);
```

Comment:
The `response.resultCode` should be checked here instead of the `error` since the request can succeed but also have an error message.

Reply:
Fixed